overlord/servicestate/quota_handlers.go

```diff
@@ -595,7 +601,37 @@ func ensureSnapServicesStateForGroup(st *state.State, grp *quota.Group, opts *en
 	return restartSnapServices(st, nil, appsToRestartBySnap, nil)
 }
 
+func ensureGroupIsNotMixed(group string, allGrps map[string]*quota.Group) error {
+	grp, ok := allGrps[group]
+	if ok {
+		if len(grp.SubGroups) != 0 && len(grp.Snaps) != 0 {
+			return fmt.Errorf("quota group %q has mixed snaps and sub-groups, which is not supported anymore. please remove either snaps or sub-groups to modify this group", group)
```

Comment:
snaps cannot be removed from sub-groups so the only remedy is to delete the group

Reply:
I updated the comment, and the spread tests should also be fixed now.